src/App.jsx

```diff
@@ -2,19 +2,31 @@ import * as React from "react";
 import TodoList from "./TodoList"; //Import TodoList component
 import AddTodoForm from "./AddTodoForm"; //Import AddTodoForm component
 
+function useSemiPersistentState(key, initialState) {
+  const [value, setValue] = React.useState(
+    JSON.parse(localStorage.getItem(key)) || initialState
+  );
+
+  React.useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(value));
+  }, [value, key]);
```

Comment:
our assignment require other variable name 
<img width="556" alt="image" src="https://github.com/user-attachments/assets/8eaada17-5e4f-4115-8877-a78741528322">

I also think our instruction asked use this pattern: 
<img width="515" alt="image" src="https://github.com/user-attachments/assets/dad274db-ffe2-4acf-888c-6d31eb8d72cc">

Yoy created great reusable hook though. Just nit as assignment asked.